### ISSUE TITLE
Reactions doesn't work if iCloudDrive is disabled.

### ISSUE THREAD

**Issue by PPacie** [Canillitapp/headlines-iOS#39]
Basically, CloudKit doesn't return the `fetchUserRecordID` if iCloudDrive is disabled in Settings even if the user is logged into iCloud.

I am not sure weather this is the way it should work or if it's an issue. Just in case I've opened a [rdar 32277653](https://openradar.appspot.com/radar?id=5027919789293568) .

[Apple documentation](https://developer.apple.com/library/content/documentation/DataManagement/Conceptual/CloudKitQuickStart/CreatingaSchemabySavingRecords/CreatingaSchemabySavingRecords.html#//apple_ref/doc/uid/TP40014987-CH3-SW12) is kind of confusing as they encourage to enable iCloudDrive but they don't specify that it is a requirement. However, from a logical point of view, this shouldn't be a requirement to get the iCloud `UserRecordID` unless you would need to make use of the disk space of iCloudDrive (e.g: save documents to iCloudDrive, etc).

**Comment by betzerra** [comment]
@PPacie great finding, thanks!

For now, all I'm going to do is the following:
- show a modal explaining that you need to be logged in at iCloud (and having iCloudDrive enabled).
- add tracking when the modal is shown.

After shipping, I'll measure the number of people that can't use reactions because of this. If the number is significant I'll add the option of logging with twitter.

**Comment by PPacie** [comment]
Yeah, I think another alternative would be to cache a local timestamp token so you avoid bothering the user with login/pass.

Only downside of local cached token would be the case where the user uninstall and re-install the app. The user wouldn't be able to recover past reactions. 

On the other hand, an Apple CloudKit engineer told me that that's the way it's intended to work (iCloudDrive should be enabled for CK to work) but that the documentation isn't clear. I suggested that it would be way much useful to be able to get the iCloudID even if iCloudDrive is disabled. Let's see what happens. Keep you updated.
